types/backbone/index.d.ts

```diff
@@ -138,7 +138,7 @@ declare namespace Backbone {
         constructor(attributes?: any, options?: any);
         initialize(attributes?: any, options?: any): void;
 
-        fetch(options?: ModelFetchOptions): JQueryXHR;
+        fetch(options?: ModelFetchOptions): JQuery.Promise<any>;
```

Comment:
Turns out this was only an issue with JQuery types v3.2.5. 3.2.6 has resolved this.